### PR TITLE
octeon: cleanup base-files

### PR DESCRIPTION
**Pull request by damex**
improvements:
 - use single awk (busybox version works well)
   command to find board name in 01_sysinfo and
   to find sysupgrade directory inside sysupgrade tar
 - use tabs and curly brackets for variables everywhere
   helps with not making a mistake in the future and
   to keep code clean-er
 - use separate condition logic to make statements certain
 - add checks for devices being block devices and that files
   are actually exist before doing anything with them.
   we need to be certainly sure that environment is right
   before doing anything to the device.
 - stop guessing which device we need to use for flashing
   since we already know that information.
 - get rid of exception in logic during flash for itus and
   make it follow same pattern with a kernel partition backup.
   we add kernel image name for it follow the same pattern.
 - reuse the code and add comments

tldr: actualized code, reuse and document it.

tested on:
  - ubnt_edgerouter-lite
  - ubnt_edgerouter
  - generic board

what tested:
  - sysupgrade with keeping configuration
  - sysupgrade without keeping configuration
  - new install
    (not applicable to ubnt_edgerouter-lite since it is done from pc)
  - generic octeon booting on generic octeonplus or higher board

Signed-off-by: Roman Kuzmitskii damex.pp@icloud.com

### CODE REVIEW

File: target/linux/octeon/base-files/lib/preinit/01_sysinfo [hunk 1]
```diff
@@ -2,34 +2,30 @@ do_sysinfo_octeon() {
...
+			name="generic"
+			;;
 	esac
 
-	[ -e "/tmp/sysinfo/" ] || mkdir -p "/tmp/sysinfo/"
+	mkdir -p "/tmp/sysinfo"
 
-	echo "$name" > /tmp/sysinfo/board_name
-	echo "$machine" > /tmp/sysinfo/model
+	echo "${name}" > /tmp/sysinfo/board_name
```
Comment (neheb) [inline target/linux/octeon/base-files/lib/preinit/01_sysinfo:27]: Why?

File: target/linux/octeon/base-files/lib/preinit/79_move_config [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
-			[ -f "/mnt/$BACKUP_FILE" ] && mv -f "/mnt/$BACKUP_FILE" /
-			umount /mnt
+			device="/dev/mmcblk1p1"
 			;;
 	esac
+
+	# checks for:
+	# 1. device variable is not empty
+	# 2. device variable points to actual block device
+	if [ ! -z "${device}" ] && [ -b "${device}" ] ; then
```
Comment (neheb) [inline target/linux/octeon/base-files/lib/preinit/79_move_config:19]: ! -z is the same as -n. Run through shellcheck.

Comment (damex) [reply target/linux/octeon/base-files/lib/preinit/79_move_config:19]: `-n` checks if variable exists. even empty.
`! -z` will also tell if it is not empty.

File: target/linux/octeon/base-files/lib/preinit/01_sysinfo (hunk 1 above)
Comment (damex) [reply target/linux/octeon/base-files/lib/preinit/01_sysinfo:27]: becuase of curly brackets.
easier to understand that this thing inside them should expand into the variable.
and less chance to screw it if someone wants to expand that call.

it is just a good practice to use them at all times then don't.

File: target/linux/octeon/base-files/lib/preinit/79_move_config (hunk 2 above)
Comment (neheb) [reply target/linux/octeon/base-files/lib/preinit/79_move_config:19]: https://github.com/koalaman/shellcheck/wiki/SC2236

Comment (damex) [reply target/linux/octeon/base-files/lib/preinit/79_move_config]: thank you.

pushed a fix resolving that.

also did `-n` with `return 1` on failure.
same thing but makes more sense in that case. also updated comments about that

File: target/linux/octeon/base-files/lib/preinit/01_sysinfo [hunk 3]
```diff
@@ -1,35 +1,33 @@
+#!/bin/sh
+
```
Comment (adschm) [inline target/linux/octeon/base-files/lib/preinit/01_sysinfo:2]: These are non-executable lib files. There should be no shebang.

File: target/linux/octeon/base-files/etc/board.d/01_network [hunk 4]
```diff
@@ -7,13 +7,15 @@
...
-	;;
+board="$(board_name)"
+
+case "${board}" in
+	itus,shield-router)
+		ucidef_set_interfaces_lan_wan "eth1 eth2" "eth0"
+		;;
+	*)
+		ucidef_set_interfaces_lan_wan "eth0" "eth1"
+		;;
```
Comment (adschm) [inline target/linux/octeon/base-files/etc/board.d/01_network:18]: I'm not a fan of this kind of cosmetic changes, at least if they happen isolated like here. There is no clear better or worse here and effectively, somebody could just change it back for the same "reason" in two years ...

Please remove this kind of changes from the commit.

File: target/linux/octeon/base-files/lib/preinit/01_sysinfo (hunk 1 above)
Comment (adschm) [reply target/linux/octeon/base-files/lib/preinit/01_sysinfo:29]: Like above, brackets or no brackets is a matter of taste here, so I prefer not changing it without a reason.

File: target/linux/octeon/base-files/lib/preinit/79_move_config [hunk 5]
```diff
@@ -1,21 +1,28 @@
-# Copyright (C) 2014 OpenWrt.org
+#!/bin/sh
```
Comment (adschm) [inline target/linux/octeon/base-files/lib/preinit/79_move_config:1]: No shebang in lib files. And please don't remove any Copyright marks.

File: target/linux/octeon/base-files/lib/preinit/79_move_config [hunk 6]
```diff
@@ -1,21 +1,28 @@
...
-			umount /mnt
+			device="/dev/mmcblk1p1"
 			;;
 	esac
+
+	# checks for:
+	# 1. device variable is not empty
+	# 2. device variable points to actual block device
+	if [ -n "${device}" ] && [ -b "${device}" ] ; then
+		. /lib/upgrade/common.sh
```
Comment (adschm) [inline target/linux/octeon/base-files/lib/preinit/79_move_config:20]: What's the benefit of moving this include?

File: target/linux/octeon/base-files/lib/upgrade/platform.sh [hunk 7]
```diff
@@ -1,123 +1,152 @@
...
-			*root=*)
-				rootfsdev="${cmdline##*root=}"
-				rootfsdev="${rootfsdev%% *}"
-			;;
-		esac
-
-		echo "${rootfsdev}"
-	fi
-}
+#!/bin/sh
```
Comment (adschm) [inline target/linux/octeon/base-files/lib/upgrade/platform.sh:1]: see above.

File: target/linux/octeon/base-files/lib/upgrade/platform.sh [hunk 8]
```diff
@@ -1,123 +1,152 @@
...
+			if [ "${kernel_length}" = 0 ] || [ "${rootfs_length}" = 0 ] ; then
+				echo "The upgrade image is corrupt."
+				return 1
+			fi
+			return 0
+			;;
 	esac
 
-	echo "Sysupgrade is not yet supported on $board."
+	echo "Sysupgrade is not yet supported on ${board}."
```
Comment (adschm) [inline target/linux/octeon/base-files/lib/upgrade/platform.sh:150]: see above.

File: target/linux/octeon/base-files/lib/preinit/79_move_config (hunk 2 above)
Comment (adschm) [reply target/linux/octeon/base-files/lib/preinit/79_move_config]: > `-n` checks if variable exists. even empty.

That's _wrong. `-n` will be _false_ for an empty string, no matter whether the variable exists or not.

File: target/linux/octeon/base-files/lib/preinit/79_move_config (hunk 6 above)
Comment (damex) [reply target/linux/octeon/base-files/lib/preinit/79_move_config:20]: it is only needed for `$BACKUP_FILE` variable and we might not even get to the point where it is needed (condition check does not succeed).

if target does not support configuration backup i.e. `generic` target - it won't get to load `/lib/upgrade/common.sh`.

this is the only reason.

File: target/linux/octeon/base-files/lib/upgrade/platform.sh (hunk 8 above)
Comment (damex) [reply target/linux/octeon/base-files/lib/upgrade/platform.sh:148]: should i use raw variables without curly brackets?

there does not seem to be a code guideline and it would preffered to keep the same style.

File: target/linux/octeon/base-files/etc/board.d/01_network (hunk 4 above)
Comment (damex) [reply target/linux/octeon/base-files/etc/board.d/01_network:18]: by 'no cosmetic changes' you mean:
- no curly brackets
- do not add tab ident where it was not there before (although openwrt use tabs ident everywhere?)

and that seems to be it?

File: target/linux/octeon/base-files/lib/preinit/01_sysinfo (hunk 3 above)
Comment (damex) [reply target/linux/octeon/base-files/lib/preinit/01_sysinfo]: thanks, got rid of shebang

File: target/linux/octeon/base-files/lib/preinit/79_move_config (hunk 5 above)
Comment (damex) [reply target/linux/octeon/base-files/lib/preinit/79_move_config]: thanks, got rid of shebang and brought back copyright mark

File: target/linux/octeon/base-files/lib/upgrade/platform.sh (hunk 7 above)
Comment (damex) [reply target/linux/octeon/base-files/lib/upgrade/platform.sh]: thanks, got rid of shebang

**Comment by damex** [comment]
i am going to split it between three separate commits:
- sysinfo logic
- move config logic
- platform update logic

**Comment by adschm** [comment]
Commit != PR

The proper way would have been to rebase and add three proper commits in this existing PR...

**Comment by damex** [comment]
@adschm sorry but i already made a PR for each of them:

octeon: improve platform upgrade
https://github.com/openwrt/openwrt/pull/3529

octeon: use dedicated function to move config backup
https://github.com/openwrt/openwrt/pull/3528

octeon: use single awk to parse octeon board name
https://github.com/openwrt/openwrt/pull/3527

want me to close them and reopen this PR instead?
